util/test/execution_limiter.py

```
@@ -0,0 +1,107 @@
+    # modified, the name of the test that has the lock, and the
+    # test's timeout value. If there's issues opening or parsing the
+    # file, return a default of: (1, "corrupted_lock_file", 0), so
+    # that the waiting test can continue.
+    def _parse_lock_file(self):
+        default_lock_file = (1, "corrupted_lock_file", 0)
+        try:
+            with open(self.lock_file, 'r') as f:
+                lines = f.read().splitlines()
+                other_filename = lines[0]
+                other_timeout = int(lines[1])
+
+                modified = time.time() - os.path.getmtime(self.lock_file)
+                return (modified, other_filename, other_timeout)
+        except:
+            log_info('Failed to parse lock file, using default one')
+            return default_lock_file
+
+    # Grab the execution "lock", spin waiting while some other test
+    # has the lock. If for some reason the other test didn't clean
+    # up the locking file, we'll continue if the lock hasn't been
+    # modified in the other test's timeout value.
+    def _lock(self):
+	modified = 0
+	other_timeout = 1
+	while os.path.exists(self.lock_file) and modified < other_timeout:
+            modified, other_filename, other_timeout = self._parse_lock_file()
+            log_info('"{0}" is waiting at most {1}s for "{2}" (started {3:.1f}s '
+                     'ago) to finish executing'.format(
+                     self.exec_name, other_timeout, other_filename, modified))
+            self._backoff_sleep()
+
+	# actually grab the "lock" by creating the file
+        log_info('Grabbing exec-lock for "{0}"'.format(self.exec_name))
+	with open(self.lock_file, 'w') as f:
+	    f.write('{0}\n{1}\n'.format(self.exec_name, self.exec_timeout))
+
+
+    def _unlock(self):
+	try:
+	    # release the "lock" by removing the file
+            log_info('Releasing exec-lock for "{0}"'.format(self.exec_name))
+	    os.remove(self.lock_file)
+	except:
+	    pass
```

Is a silent error upon failure to unlock acceptable? Is this covering the case when the file is already removed, or something else?